tests/testthat/test-check_country_class.R

```diff
@@ -0,0 +1,51 @@
+check_country_fields <- function(regionClass) {
+  expected_public_fields <- names(CountryTemplate$public_fields)
+  region_public_fields <- names(regionClass$public_fields)
+  for (field in expected_public_fields) {
+    test_that(
+      paste("Expect", field, "is in", class(regionClass)[1], "public fields"),
+      {
+        expect_true(field %in% region_public_fields)
+      }
+    )
+  }
+  return(0)
+}
+
+check_all_country_fields <- function() {
+  avalibale_country_data <- get_available_datasets()
+  avaliable_sources <- avalibale_country_data$country
+  for (country in avaliable_sources) {
+    regionClass <- get(country)
+    check_country_fields(regionClass)
+  }
+  return(0)
+}
+
+check_country_methods <- function(regionClass) {
+  expected_public_methods <- names(CountryTemplate$public_methods)
+  region_public_methods <- names(regionClass$public_methods)
+  for (method in expected_public_methods) {
+    test_that(
+      paste("Expect", method, "is in", class(regionClass)[1], "public methods"),
+      {
+        expect_true(method %in% region_public_methods)
+      }
+    )
+  }
+  return(0)
+}
+
+check_all_country_methods <- function() {
+  avalibale_country_data <- get_available_datasets()
```

Comment:
curse of available strikes again 😉 

It might be better to get all the datasets and then operate on them individually (as in the previous implementation). This will be helpful when we want to just test as single dataset downloads and passes tests. There are other ways this could be done here but I'm not sure how they would avoid the code repetition.

Reply:
My brain knows it, my fingers do not :( 

Think your'e going to have to start fining me for this. Every misspell = 1 awful task!